onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -0,0 +1,290 @@
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<uint16_t>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "INT16") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<int16_t>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "INT32") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<int32_t>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "INT64") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<int64_t>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "BOOL") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<bool>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "FP16") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<float>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "FP64") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<double>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "UINT32") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<uint32_t>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "UINT64") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<uint64_t>(), TensorShape{dim}, cpu_allocator_);
+  } else if (data_type == "BF16") {
+    return std::make_unique<Tensor>(onnxruntime::DataTypeImpl::GetType<BFloat16>(), TensorShape{dim}, cpu_allocator_);
+  } else {
+    return {};
+  }
+}
+
+TritonInvoker::TritonInvoker(const CloudEndPointConfig& config) : CloudEndPointInvoker(config) {
+  if (ReadConfig(kCloudUri, uri_) &&
+      ReadConfig(kCloudModelName, model_name_) &&
+      ReadConfig(kCloudModelVer, model_ver_, false) &&
+      ReadConfig(kCloudVerbose, verbose_, false)) {
+    if (tc::InferenceServerHttpClient::Create(&triton_client_, uri_, verbose_).IsOk()) {
+      cpu_allocator_ = std::make_shared<CPUAllocator>();
+    } else {
+      status_ = ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to initialize triton client");
+    }
+  }
+}
+
+onnxruntime::Status TritonInvoker::Send(const CloudEndPointConfig& run_options,
+                                        const InlinedVector<std::string>& input_names,
+                                        gsl::span<const OrtValue> ort_inputs,
+                                        const InlinedVector<std::string>& output_names,
+                                        std::vector<OrtValue>& ort_outputs) const noexcept {
+  if (!status_.IsOK()) return status_;
+  if (run_options.count(kCloudAuthKey) == 0) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                           "auth key must be specified for triton client");
+  }
+
+  tc::Headers http_headers;
+  http_headers["Authorization"] = std::string{"Bearer "} + run_options.at(kCloudAuthKey);
+
+  if (ort_inputs.size() != input_names.size()) {
```

Comment:
nit: Input validation can be moved up before doing any real work.

Reply:
Moved.